server/etcdserver/version/monitor.go

```diff
@@ -73,6 +79,24 @@ func (m *Monitor) UpdateClusterVersionIfNeeded() {
 	}
 }
 
+// UpdateStorageVersionIfNeeded updates the storage version if it differs from cluster version.
+func (m *Monitor) UpdateStorageVersionIfNeeded() {
+	cv := m.s.GetClusterVersion()
+	if cv == nil {
+		return
+	}
+	m.s.Lock()
```

Comment:
Do we need this lock for GetStorageVersion? Or could we move to the if function below?

Reply:
Here I wanted to make sure that GetStorageVersion and UpdateStorageVersion is done in same transaction.